src/main/java/de/azapps/kafkabackup/common/offset/OffsetSink.java

```
@@ -80,28 +80,26 @@ public void syncOffsets() {
-            // and catch the exception that occurs if the consumer group is missing from the TopicPartition's committed offsets.
-            // TODO: fix when 2.4.1 is released!
-            for (TopicPartition tp : partitions) {
-                ListConsumerGroupOffsetsOptions listConsumerGroupOffsetsOptions = new ListConsumerGroupOffsetsOptions();
-                listConsumerGroupOffsetsOptions.topicPartitions(Collections.singletonList(tp));
-                try {
-                    Map<TopicPartition, OffsetAndMetadata> offsetsAndMetadata = adminClient
-                            .listConsumerGroupOffsets(consumerGroup, listConsumerGroupOffsetsOptions)
-                            .partitionsToOffsetAndMetadata()
-                            .get();
-                    partitionOffsetsAndMetadata.put(tp, offsetsAndMetadata.get(tp));
-                } catch (ExecutionException e) {
-                    log.debug("No committed offsets for consumer group {} on topic {} partition {}", consumerGroup, tp.topic(), tp.partition());
-                } catch (InterruptedException e) {
-                    throw new RetriableException(e);
-                }
-            }
+            ListConsumerGroupOffsetsOptions listConsumerGroupOffsetsOptions = new ListConsumerGroupOffsetsOptions();
+            listConsumerGroupOffsetsOptions.topicPartitions((List<TopicPartition>)partitions);
+
+            partitionOffsetsAndMetadata = adminClient
+                .listConsumerGroupOffsets(consumerGroup, listConsumerGroupOffsetsOptions)
+                .partitionsToOffsetAndMetadata()
+                .get()
+                .entrySet()
+                .stream()
+                .filter(map -> map.getValue() != null)
+                .collect(Collectors.toMap(map -> map.getKey(), map -> map.getValue()));
+
+        } catch (InterruptedException | ExecutionException e) {
```

Is it possible to determine from the docs what would constitute and ExecutionException? In the old version the null offset caused an ExecutionException, and that wasn't "Retryable" since it would always fail. That shouldn't happen anymore in 2.4.1, but perhaps there are other non-retryable things that could happen?

---

Can't find much except this:

> Exception thrown when attempting to retrieve the result of a task that aborted by throwing an exception. This exception can be inspected using the getCause() method.

I changed it to an error thrown instead of log because if we get the exception we will have no data to save. Do you think that make sense?